### ISSUE TITLE
Changes to be included in Release 0.5.1

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#886]
I'm currently preparing the 0.5 release.

This issue is for creating a list of all pull requests / commits we want to include in the 0.5.1 release.
I want that the 0.5.1 release contains only bugfixes. 

I will create a release-0.5.1 branch that I'll manually maintain by cherry picking commits from `master`.

Fixes included in 0.5.1:
- https://github.com/stratosphere/stratosphere/commit/73a9dcc92ae8911bfc7dd429fba78be6a63e3043 (from PR https://github.com/stratosphere/stratosphere/pull/876) 

Commits to be included into the 0.5.1 release:
- https://github.com/stratosphere/stratosphere/pull/884
- https://github.com/stratosphere/stratosphere/issues/887
- https://github.com/stratosphere/stratosphere/pull/850

**Comment by fhueske** [comment]
Why not simply assigning issues and PRs to milestones?
I find this much more comfortable than "meta-issues" because you can filter
issues (and PRs) for milestones.

2014-05-29 14:37 GMT+02:00 Robert Metzger notifications@github.com:

**Comment by rmetzger** [comment]
I agree that it is more comfortable to use the appropriate feature for that.
The problem with the current 0.5.1 milestone is that it contains also features (such as https://github.com/stratosphere/stratosphere/issues/760).
If you want, I can clean up the 0.5.1 milestone to let it only contain bug fixes and move all features to 0.6.

**Comment by fhueske** [comment]
OK, go ahead, but we should then start to work on different branches for
different releases.
We cannot delay merging feature PRs because of bugfix releases...

2014-05-29 14:50 GMT+02:00 Robert Metzger notifications@github.com:

**Comment by uce** [comment]
I agree with @fhueske. @rmetzger, that's a good idea then. :+1:

**Comment by rmetzger** [comment]
I'll change the version in the master to 0.6-SNAPSHOT in the next few minutes.
My understanding of everything is the following:
- `master` contains the code of the current SNAPSHOT version, for the next major feature release
- `release-0.x.y` contains the specific version of a release.

**Comment by uce** [comment]
We work on `master` and cherry-pick bug fixes to `release-0.x.y`, right?

**Comment by rmetzger** [comment]
Exactly. At least I'll try that. I'm not sure how soon it will start creating problems. My hope is also that the bugfixes are not too big.

**Comment by rmetzger** [comment]
I created the 0.5.1 branch https://github.com/stratosphere/stratosphere/tree/release-0.5.1
